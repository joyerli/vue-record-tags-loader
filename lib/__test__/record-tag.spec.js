const fs = require('fs');
const mockContext = require('./mock-context');
const loader = require('../index');
const store = require('../store');

it('记录vue文件组件中的所有的标签', async () => {
  store.clear();

  const callbackTplSpy = jest.fn();
  const context = mockContext({
    callbackSpy: callbackTplSpy,
    resourcePath: 'path/to/plain.vue',
  });
  const code = fs.readFileSync('lib/__test__/plain.vue', {
    encoding: 'utf8',
  });

  loader.call(context, code);

  expect(store.get('path/to/plain.vue')).toMatchSnapshot();
});

it('支持忽略文件', async () => {
  store.clear();

  const callbackTplSpy = jest.fn();
  const context = mockContext({
    callbackSpy: callbackTplSpy,
    resourcePath: 'path/to/plain.vue',
    exclude: ['path/to/plain.vue'],
  });
  const code = fs.readFileSync('lib/__test__/plain.vue', {
    encoding: 'utf8',
  });

  loader.call(context, code);

  expect(store.get('path/to/plain.vue')).toMatchSnapshot();
});

it('屏蔽js中标签的影响', async () => {
  store.clear();

  const callbackTplSpy = jest.fn();
  const context = mockContext({
    callbackSpy: callbackTplSpy,
    resourcePath: 'path/to/with-js.vue',
    resourceQuery: '?vue&type=script&id=21aa4478&',
  });
  const code = fs.readFileSync('lib/__test__/with-js.vue', {
    encoding: 'utf8',
  });

  loader.call(context, code);

  expect(store.get('path/to/with-js.vue')).toMatchSnapshot();
});
