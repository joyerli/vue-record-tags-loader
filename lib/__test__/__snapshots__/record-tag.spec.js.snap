// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`屏蔽js中标签的影响 1`] = `null`;

exports[`支持忽略文件 1`] = `null`;

exports[`记录vue文件组件中的所有的标签 1`] = `
Array [
  "style",
  "template",
  "section",
  "cd-panel",
  "el-row",
  "el-col",
  "mkt-icon",
  "cd-text-action",
  "span",
  "el-form",
  "el-form-item",
  "cd-input-money",
  "div",
  "cd-button",
  "CdButton",
  "md-agreement",
  "script",
]
`;
